room.go

```diff
@@ -69,7 +70,7 @@ func (r *Room) consumeJoiners() {
 
 func (r *Room) consumeLeavers() {
```

Comment:
Might need a single loop to process joining and leaving. Otherwise, there could be a race.

FYI, you can run `go test -race` and it will find possible race conditions. https://golang.org/doc/articles/race_detector.html